include/irr/asset/filters/CBlitImageFilter.h

```diff
@@ -416,6 +429,10 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<KernelX,KernelX,Ke
 								auto sample = lineBuffer+i*MaxChannels;
 								decodePixels<value_type>(inFormat,srcPix,sample,inBlockCoord.x,inBlockCoord.y);
 
+								value_type swizzledSample[MaxChannels];
+								state->defaultSwizzle(sample, swizzledSample, MaxChannels);
+								memcpy(sample, swizzledSample, sizeof(value_type) * MaxChannels);
```

Comment:
we don't need to memcpy, I think `decodePixels` has a variant that takes a swizzle.

Reply:
You were not to force me to split what's inside decodepixels otherwise

Reply:
but i will check it

Reply:
`decodePixels` doesn't have such variant - `convertColor` does, but we don't want it

Reply:
not an issue anymore